lib/engine/game/g_1825/game.rb

```diff
@@ -1109,11 +1109,21 @@ def check_distance(route, visits)
 
           node_hexes = {}
           visits.each do |node|
-            raise GameError, 'Cannot visit multiple towns/cities in same hex' if node_hexes[node.hex]
+            # For 3T and 4T trains, allow multiple towns in same hex but not cities
+            if %w[3T 4T].include?(route.train.name)
+              raise GameError, 'Cannot visit multiple cities in same hex' if node.city? && node_hexes[node.hex]
 
-            node_hexes[node.hex] = true
+              node_hexes[node.hex] = true if node.city?
```

Comment:
There is at least one tile with both a town and a city on it. This test is going to allow the route to visit both of these – is that right? Or is the route only allowed to visit two stops on the same tile if both are towns?

Reply:
@ollybh quite right. It's illegal unless it's a "T" train and both are dits.